optuna/storages/_heartbeat.py

```diff
@@ -0,0 +1,32 @@
+import copy
+
+import optuna
+from optuna._experimental import experimental
+
+
+@experimental("2.9.0")
+def fail_stale_trials_with_callback(study: "optuna.Study") -> None:
```

Comment:
Name is open for discussion. Just, `fail_stale_trials` might be simpler and better.

Reply:
I agree. With `with_callback`, it sounds like a callback is needed.

Reply:
Thanks for the input. I remove the "with_callback" part again. I agree that this "with" preposition was rather confusing.